src/spurt/io/_three_d.py

```diff
@@ -0,0 +1,263 @@
+            clsname = type(self).__name__
+            errmsg = (
+                f"{clsname} only supports 2D arrays. Unsupported shape {data.shape}"
+            )
+            raise ValueError(errmsg)
+
+        if xy.ndim != 2:
+            clsname = type(self).__name__
+            errmsg = f"{clsname} accepts xy as a 2D array. Unsupported shape {xy.shape}"
+            raise ValueError(errmsg)
+
+        # Check if time_dim is within limits
+        if time_dim >= data.ndim:
+            clsname = type(self).__name__
+            errmsg = f"{clsname} received time dimension outside array dimensions"
+            raise ValueError(errmsg)
+
+        if xy.shape[0] != data.shape[1 - time_dim]:
+            clsname = type(self).__name__
+            errmsg = (
+                f"{clsname} mismatch of data and xy arrays"
+                f"{data.shape} vs {xy.shape} with time axis {time_dim}"
+            )
+            raise ValueError(errmsg)
+
+        # We only hold a reference to the data
+        self._data = data
+        self._time_dim = time_dim
+        self._xy = xy
+
+    @property
+    def dtype(self) -> DTypeLike:
+        return self._data.dtype
+
+    @property
+    def ndim(self) -> int:
+        return 2
+
+    @property
+    def time_dim(self) -> int:
+        return self._time_dim
+
+    @property
+    def space_dim(self) -> int:
+        return int(1 - self._time_dim)
+
+    @property
+    def _space_size(self) -> int:
+        return self._data.shape[self.space_dim]
+
+    @property
+    def _time_size(self) -> int:
+        return self._data.shape[self._time_dim]
+
+    @property
+    def shape(self) -> tuple[int, ...]:
+        return self._data.shape
+
+    @property
+    def xcoords(self) -> ArrayLike:
+        """Return the x-coordinate of the points as an array."""
+        return self._xy[:, slice(0, 1)]
+
+    @property
+    def ycoords(self) -> ArrayLike:
+        """Return the y-coordinate of the points as an array."""
+        return self._xy[:, slice(1, 2)]
+
+    def get_time_slice(self, key: int) -> ArrayLike:
+        """Read a block of data in time.
+
+        Only makes a copy if absolutely needed.
+
+        Parameters
+        ----------
+        key: int
+            Index on the time axis
+        """
+        ind = tuple(
+            slice(None) if ii != self._time_dim else key
+            for ii in range(self._data.ndim)
+        )
+        return self._data[ind].ravel()  # type: ignore[arg-type, index]
+
+    def get_spatial_slice(self, key: int) -> ArrayLike:
+        """Read a block of data in space.
+
+        Only makes a copy if absolutely needed.
+
+        Parameters
+        ----------
+        key: int
+            Index on the space axis
+        """
+        ind = tuple(
+            slice(None) if ii != self.space_dim else key
+            for ii in range(self._data.ndim)
+        )
+        return self._data[tuple(ind)].ravel()  # type: ignore[arg-type, index]
```

Comment:
`ind` is already a tuple

```suggestion
        return self._data[ind].ravel()  # type: ignore[arg-type, index]
```